jinahub/indexers/query/vector/FaissIndexer/__init__.py

```diff
@@ -114,36 +174,60 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
         index.nprobe = self.nprobe
         return index
 
-    @batching
     def _build_partial_index(self, vecs: 'np.ndarray', index):
         vecs = vecs.astype(np.float32)
         if self.normalize:
             from faiss import normalize_L2
+
             normalize_L2(vecs)
         index.add(vecs)
 
-    def query(self, vecs: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.ndarray', 'np.ndarray']:
+        return
+
+    @requests(on='/search')
+    def query(self, docs: DocumentArray, parameters: Dict = None, *args, **kwargs):
         """Find the top-k vectors with smallest ``metric`` and return their ids in ascending order.
-        :param keys: numpy array containing vectors to search for
-        :param top_k: upper limit of responses for each search vector
+
+        :param docs: the DocumentArray containing the documents to search with
+        :param parameters:
         """
+        if parameters is None:
+            parameters = {}
+
+        top_k = parameters.get('top_k', self.default_top_k)
+        traversal_path = parameters.get('traversal_path', self.traverse_path)
+
+        docs_to_traverse = docs.traverse_flat(traversal_path)
```

Comment:
`docs_to_index` seems a better naming? since the `tree` will already be traversed?

Reply:
You don't index these, you query with them